src/Streams.Core/Streams.fs

```diff
@@ -4,20 +4,18 @@ open System.Collections.Generic
 open System.Threading
 
 /// Provides on-demand iteration 
-type Iterator = {
+type (* internal *) Iterator = 
     /// Function for on-demand processing
-    TryAdvance : unit -> bool 
+    abstract TryAdvance : unit -> bool 
     /// Cleanup function
-    Dispose : unit -> unit 
-}
+    abstract Dispose : unit -> unit 
 
 /// Provides functions for iteration
-type Iterable = {
+type (* internal *) Iterable = 
```

Comment:
I'm preparing a PR for this, so that the contents of every Iterable looks like

    Stream(fun ... -> 
        let bulk() = ...
        let iterator() = ...
       { new Iterable with ... } )

Cheers
don